### ISSUE TITLE
Placeholder in input field is displayed at the wrong location

### ISSUE THREAD

**Issue by koendehondt** [guillep/pharo-core#15872]
**Bug description**
When using a placeholder in a `SpTextInputFieldPresenter`, the placeholder is not displayed at the position of the text.

**To Reproduce**
The problem is easily shown when opening a window like this (the list is not relevant, it is only here as an example):

```smalltalk
| box label |
label := SpTextInputFieldPresenter new
	text: 'Test';
	placeholder: 'Search';
	yourself.
box := SpBoxLayout newTopToBottom
	add: label;
	add: SpListPresenter new;
	yourself.
SpPresenter new layout: box; open
```

When the window opens, the text is positioned as expected:

<img width="417" alt="Input field - text" src="https://github.com/pharo-project/pharo/assets/753271/3e80d2ef-fab6-41be-a5d1-e3a59bc6ec8d">

However, when the text is removed and the placeholder appears, it appears at an unexpected position.

<img width="419" alt="Input field - placeholder" src="https://github.com/pharo-project/pharo/assets/753271/fd4df641-1b53-4e3e-8d2e-2d18ab7f70b1">

**Expected behavior**
The expected behaviour is that the placeholder is shown at the same position where the text is shown.

**Version information:**
 - OS: MacOS Sonoma
 - Version: 14.1.2
 - Pharo Version: 11

**Expected development cost**
No idea.

**Additional context**
None.

**Comment by Ducasse** [comment]
Thanks! I tagged it for @estebanlm

**Comment by astares** [comment]
Ideally we have a placeholder position #center, #topLeft, topRight, bottomLeft ...  to place  freely. Think of BIDI languages reading RTL instead of LTR

**Comment by estebanlm** [comment]
is hard to see how a text input (which just accepts one line of text) needs to be shown as it is shown in the example.
However, indeed the placeholder problem is there... but I already spent two weeks last year trying to fix it  (because is a morphic problem, not spec) and  I failed. 
I do not think I will have time to take this issue again anytime soon, and I do not think is a priority (since as I said, a text input should have just one line)...

**Comment by koendehondt** [comment]
@astares

> Ideally we have a placeholder position #center, #topLeft, topRight, bottomLeft ... to place freely. Think of BIDI languages reading RTL instead of LTR

I do not think so. I do not see any use case where the position of the placeholder should be different from the position of the text (whether it is RTL or LTR). There is no need to introduce extra placeholder position settings.

**Comment by koendehondt** [comment]
@estebanlm

I understand that it is not urgent. It is not urgent to me. I reported it to record the bug.

> is hard to see how a text input (which just accepts one line of text) needs to be shown as it is shown in the example.

I used the example to make it very clear what the issue is. Even when using an input field that is as high as one line of text, the problem is very visible. Consider this code:

```smalltalk
| box input |
input := SpTextInputFieldPresenter new
	text: 'Test';
	placeholder: 'Search';
	yourself.
box := SpBoxLayout newTopToBottom
	add: input expand: false;
	add: SpListPresenter new;
	yourself.
SpPresenter new layout: box; open
```

I opened it twice and cleared the input in one. As you can see the baselines of the text and the placeholder are different. I expect them to be the same. I drew a red line on the screenshot to make it clear.

<img width="514" alt="Input field - one line and placeholder" src="https://github.com/pharo-project/pharo/assets/753271/a41e786a-ce6e-4904-b9a0-fe6db018f4cb">

As you can tell from the screenshot below, even the text is not positioned correctly in the default height of the box (default because `expand: false`). It is not middle aligned in the surrounding box. That is also clear from the insertion point, which touches the top of the box, but not the bottom.

<img width="423" alt="Input field - text not middle aligned" src="https://github.com/pharo-project/pharo/assets/753271/8a8bc890-f7c4-40f1-927a-f6f40871e675">

**Comment by estebanlm** [comment]
yeah, I know the problem perfectly, believe me... :P   
I fought against that but as I said, I spent two weeks trying to understand how to align the text to the center of the input text (which is the real problem) and I failed. Morphic is very annoying sometimes...
